distribution/manifest.go

```diff
@@ -197,19 +196,12 @@ func detectManifestBlobMediaType(dt []byte) (string, error) {
 			return "", fmt.Errorf(`media-type: %q should not have "config", "layers", or "fsLayers"`, mfst.MediaType)
 		}
 		return mfst.MediaType, nil
-	case schema1.MediaTypeManifest:
-		if mfst.Manifests != nil || mfst.Layers != nil {
-			return "", fmt.Errorf(`media-type: %q should not have "manifests" or "layers"`, mfst.MediaType)
-		}
-		return mfst.MediaType, nil
```

Comment:
I rebased this after https://github.com/moby/moby/pull/43045 was merged, could use some extra eyes to double check if changes in this file are good; my intent was to get rid of the `schema1` import, but in this part of the code is also means that we skip this check, and return `MediaType` as-is.

I _guess_ that's not an issue (as we removed other code to handle these manifests), but thought it was good to at least mention it here 😅